keps/sig-node/2008-checkpoint-restore/README.md

```diff
@@ -0,0 +1,609 @@
+   component (e.g. the kernel for a security update) without shutting down
+   containers. Containers are checkpointed, system can be rebooted and
+   containers are then restored from the checkpoint.
+
+ * Fast startup: In order to quickly startup containers it is possible
+   to start containers from checkpoints. This is especially interesting
+   for containers that require a long time to initialize. Either for
+   loading many libraries or for filling large caches.
+
+ * Container migration: Migrate a running container from one host to
+   another host without losing the container state.
+
+Container migration can be seen in different container environments. The following
+gives an overview of CRIU integration to support container migration. As this
+KEP would not be possible without CRIU OpenVZ needs to be mentioned:
+
+ * CRIU has been written to support container migration in OpenVZ, which
+   ensures that CRIU based migration has been designed with containers
+   in mind:
+   * https://wiki.openvz.org/Checkpointing_and_live_migration
+
+ * LXC/LXD also provides the possibility to migrate containers from one
+   host to another: `lxc move <container> <remote>:<container>`
+
+   In addition to the simplest form of container migration (checkpoint,
+   transfer, restore) LXD also supports optimizations to decrease the
+   downtime during migration by using CRIU's pre-copy migration support.
+
+   * https://archive.fosdem.org/2018/schedule/event/containers_optimized_migration/
+   * https://lisas.de/~adrian/posts/2017-Dec-06-optimizing-live-container-migration-in-lxd.html
+
+ * Borg uses CRIU to live migrate containers between hosts to free up resources
+   on hosts which are missing resources under load:
+
+   * [Task Migration at Google Using CRIU](https://www.linuxplumbersconf.org/event/2/contributions/209/)
+   * [Update on Task Migration at Google Using CRIU](https://linuxplumbersconf.org/event/4/contributions/508/)
+
+   See especially the second presentation for limitations Google is
+   experiencing using CRIU based container migration in production.
+
+ * Podman supports container migration in its simplest form (checkpoint,
+   transfer, restore): https://criu.org/Podman
+
+The motivation is to get closer to container migration by taking the first
+step and providing support for simple checkpoint and restore. The motivation
+is existing checkpoint/restore/migration support in other container
+environments showing that it is useful and production ready.
+
+### Goals
+
+The goal of this KEP is to introduce *checkpoint* and *restore* to the CRI API.
+
+In the first step only *checkpoint* should be introduced to CRI API with
+the User Story 1 in mind (forensic use case).
+
+### Non-Goals
+
+Out of scope of this KEP are high level discussions about how to implement
+container migration. This is only about low level primitives to add
+*checkpoint* and *restore* to the API.
+
+## Proposal
+
+### Proof of Concept Implementation
```

Comment:
This section could be moved down to implementation and should include https://github.com/kubernetes/kubernetes/pull/104907

Reply:
Done.